{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "R[write to console]: ✔ ggplot2 3.3.6     ✔ purrr   0.3.4\n",
      "✔ tibble  3.1.7     ✔ dplyr   1.0.9\n",
      "✔ tidyr   1.2.0     ✔ stringr 1.4.0\n",
      "✔ readr   2.1.2     ✔ forcats 0.5.1\n",
      "\n",
      "R[write to console]: ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "library(readxl)\n",
    "library(treemapify)\n",
    "library(treemap)\n",
    "library(tidyverse)\n",
    "library(cowplot)\n",
    "library(dplyr)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 7537 Columns: 25\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "Delimiter: \",\"\n",
      "chr  (8): qseqid, sseqid, component, cinful_id, pephash, sample, contig, seq\n",
      "dbl (13): pident, length, mismatch, gapopen, qstart, qend, sstart, send, eva...\n",
      "lgl  (4): hmmerHit, verified, allStandardAA, signalMatch\n",
      "\n",
      "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
      "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "%%R -o best_hits\n",
    "best_hits <- read_csv(\"data/best_hits_Touchon2020_2022.08.15.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "The primary goal here is to understand the diversity of identified microcins in _E. coli_ genome assemblies. In this regard, we split assemblies into two classes:\n",
    "\n",
    "2. Verified microcins \n",
    "   - These assemblies had a protein with the exact same sequence as the verified microcin sequences known in the literature\n",
    "3. novel\n",
    "   - These assemblies contained a match using BLAST and HMM to a verified microcin, though it is agnostic of how well the match is or any other components that were identified in the assembly.\n",
    "\n",
    "What we will do is make a treemap to make two rectangles whose areas correspond to the number of assemblies that belong to each of these two groups. Verified and novel sequences zones contain various different protein sequences, and so will have sub rectangles whose areas correspond the amount of assemblies that had a given protein sequence.\n",
    "\n",
    "## Data to be visualized\n",
    "This contains, the sequence, unique identifier, whether or not it was verified."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of genomes with PUTATIVE microcins identified by Cinful:\"\n",
      "[1] 313\n",
      "[1] \"Number of unique peptides identified as microcins by Cinful:\"\n",
      "[1] 101\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "n_genome <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    count(sample) %>%\n",
    "    nrow()\n",
    "\n",
    "n_put <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    count(pephash) %>%\n",
    "    nrow()\n",
    "\n",
    "print(\"Number of genomes with PUTATIVE microcins identified by Cinful:\")\n",
    "print(n_genome)\n",
    "\n",
    "print(\"Number of unique peptides identified as microcins by Cinful:\")\n",
    "print(n_put)\n",
    "#Number of putative microcins identified (no check for HMMer hits)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of TOTAL sample genomes with HMMer verified microcins:\"\n",
      "[1] 288\n",
      "[1] \"Number of unique microcin sequences verified by HMMer:\"\n",
      "[1] 64\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "count_of_hmmer_hits <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    count(sample) %>%\n",
    "    nrow()\n",
    "\n",
    "\n",
    "tot_uniq_hmmer_peptides <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    count(pephash) %>%\n",
    "    nrow()\n",
    "\n",
    "print(\"Number of TOTAL sample genomes with HMMer verified microcins:\")\n",
    "print(count_of_hmmer_hits)\n",
    "\n",
    "print(\"Number of unique microcin sequences verified by HMMer:\")\n",
    "print(tot_uniq_hmmer_peptides)\n",
    "\n",
    "#Number of genomes with microcins that pass HMMer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of sample genomes with KNOWN microcins:\"\n",
      "[1] 211\n",
      "[1] \"Number of KNOWN microcins identified:\"\n",
      "[1] 6\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "\n",
    "n_ver_genomes <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    filter(verified == TRUE) %>%\n",
    "    count(sample) %>%\n",
    "    nrow()\n",
    "\n",
    "print(\"Number of sample genomes with KNOWN microcins:\")\n",
    "print(n_ver_genomes)\n",
    "\n",
    "verified_microcin_peptides <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    filter(verified == TRUE) %>%\n",
    "    count(pephash) %>%\n",
    "    nrow()\n",
    "\n",
    "print(\"Number of KNOWN microcins identified:\")\n",
    "print(verified_microcin_peptides)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of sample genomes with UNKNOWN microcins:\"\n",
      "[1] 113\n",
      "[1] \"Number of UNKNOWN microcins identified:\"\n",
      "[1] 58\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "n_unk_genomes <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    filter(verified == FALSE) %>%\n",
    "    count(sample) %>%\n",
    "    nrow()\n",
    "\n",
    "print(\"Number of sample genomes with UNKNOWN microcins:\")\n",
    "print(n_unk_genomes)\n",
    "\n",
    "n_unk_microcins <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    filter(verified == FALSE) %>%\n",
    "    count(pephash) %>%\n",
    "    nrow()\n",
    "\n",
    "print(\"Number of UNKNOWN microcins identified:\")\n",
    "print(n_unk_microcins)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total genome assemblies analyzed:\"\n",
      "[1] 1224\n",
      "[1] \"Samples with HMMER hits:\"\n",
      "[1] 288\n",
      "[1] \"Samples without HMMER hits:\"\n",
      "[1] 936\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "num_samples <- nrow(unique(best_hits[,\"sample\"]))\n",
    "\n",
    "samples_wo_HMMER_microcins <- num_samples - as.integer(count_of_hmmer_hits)\n",
    "\n",
    "print(\"Total genome assemblies analyzed:\")\n",
    "print(num_samples)\n",
    "print(\"Samples with HMMER hits:\")\n",
    "print(count_of_hmmer_hits)\n",
    "print(\"Samples without HMMER hits:\")\n",
    "print(samples_wo_HMMER_microcins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 288\n",
      "[1] 211\n",
      "[1] 113\n",
      "[1] 36\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Total genomes with hmmer-verified microcins \n",
    "A <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    count(sample)\n",
    "A\n",
    "#288\n",
    "\n",
    "#Total genomes with hmmer-verified microcins that are a 100% match to previously known microcins\n",
    "B <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    filter(verified == TRUE) %>%\n",
    "    count(sample)\n",
    "B\n",
    "#211\n",
    "\n",
    "#Total genomes with hmmer-verified microcins that do not match previously known microcins (novel)\n",
    "C <- best_hits %>%\n",
    "    filter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "    filter(verified == FALSE) %>%\n",
    "    count(sample)\n",
    "C\n",
    "#113\n",
    "\n",
    "#Total genomes with both previously known and novel microcins\n",
    "inner_join(B[,1], C[,1], by = \"sample\")\n",
    "\n",
    "print(nrow(A))\n",
    "print(nrow(B))\n",
    "print(nrow(C))\n",
    "print(nrow(inner_join(B[,1], C[,1], by = \"sample\")))\n",
    "\n",
    "\n",
    "#    write_csv(\"data/supplemental/count_of_6_verified_microcins.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'pephash', 'verified', 'sample'. You can\n",
      "override using the `.groups` argument.\n",
      "# A tibble: 365 × 5\n",
      "   pephash                                           verified sample seq       n\n",
      "   <chr>                                             <fct>    <chr>  <chr> <int>\n",
      " 1 v1_PLS_030767dd6c0e5686eef2a8cbf65d305c761e4b8fc… FALSE    GCA_9… MREI…     1\n",
      " 2 v1_PLS_030767dd6c0e5686eef2a8cbf65d305c761e4b8fc… FALSE    GCA_9… MREI…     1\n",
      " 3 v1_PLS_0719676e795eaed3af7fb25fcc33805c9c4a26902… FALSE    GCA_9… ITLN…     1\n",
      " 4 v1_PLS_162b4d1d19ac7e08ea0998616fd4a878ff9bbec04… FALSE    GCA_9… MRTL…     1\n",
      " 5 v1_PLS_17d83a4d72fa714b8588985e84dc7828d8113227b… FALSE    GCA_9… YVFF…     1\n",
      " 6 v1_PLS_1a8057dff938353b9638f6258b4cf9f4f34d47a3b… FALSE    GCA_9… MVKR…     1\n",
      " 7 v1_PLS_1a8057dff938353b9638f6258b4cf9f4f34d47a3b… FALSE    GCA_9… MVKR…     1\n",
      " 8 v1_PLS_1cddbead564a238a4068fa38deea45eb3eccc207d… FALSE    GCA_9… MASI…     1\n",
      " 9 v1_PLS_2201698b4769e017fab30bcefc9e64ce3bb721713… FALSE    GCA_9… MRNI…     1\n",
      "10 v1_PLS_2201698b4769e017fab30bcefc9e64ce3bb721713… FALSE    GCA_9… MRNI…     1\n",
      "# … with 355 more rows\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "treemap_data <- best_hits %>%\n",
    "\tfilter(component == \"microcins.verified\") %>%\n",
    "    filter(hmmerHit == TRUE) %>%\n",
    "\tgroup_by(pephash,verified,sample,seq) %>% summarise(n = n()) %>%\n",
    "\tungroup() %>%\n",
    "\tmutate(verified = as.factor(verified)) %>%\n",
    "#\tadd_row(pephash=NA, verified=\"no_microcin\", n=samples_wo_microcins) %>%\n",
    "\twrite_csv(\"data/supplemental/Ecoli_microcin_treemap_data.csv\")\n",
    "\n",
    "treemap_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## Treemap generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'pephash', 'verified'. You can override\n",
      "using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAyAAAAGQCAIAAADZR5NjAAAWJUlEQVR4nO3d629c95nY8d+ZMzPk8E5J1M26WCvJlixbvkW+JG6iJJugQdIiu0jhTdHAvQT5F4oCyau8XLRo3xdF4b4ysECMNIvdYtOiKTa33Vwcx/bKsiyZkqg7JYp3cjgzfcFtVnGdmCM+Z4Y0P59XBnzmOQ8owvr6zJkzWavVSgAAxCl1ewEAgI8agQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFrChLS0tZVn26quvHjx4sFarPffcc2fOnOn2UgAfImu1Wt3eAeB3Wlpa6u3tPXbs2Msvv1yr1V566aX9+/e/+uqr3d4L4PdxBQvYBL75zW+eOnXq0Ucf/cY3vjE+Pt7tdQA+hMACNoGTJ0+u/sPIyEh3NwFYC4EFbALVarXbKwC0QWABAAQTWAAAwQQWAEAwj2kAAAjmChYAQDCBBQAQTGABAAQTWAAAwQQWAECw8hqP+9a3vvXKK680Go1CtwEgpTQ9PT03Nzc2NlYur/W/0kDR8jx/8cUXv/3tb6/p6NbaDA0NFbw2AMCGNjQ0tMZwWuv/GzWbzZTS11/8Z8cffLDAxYGPivLgYO/Yrsbi4sLE5W7vsvn823//H1YajUcff/IffeGPu73L5jPUV/2DXSML9ZW3r97u9i6F2D5Y2z86eGducfzmdLd3aduJfTsq5dLZK3fml+vd3qU9E+ff+e4rL6/m0Fq0d/H59BOPf+apJ9rfCthyanv3jTz6+PLt25M/+XG3d9l8/t1//E+p0diz7+BTn/mn3d5l8zk4NvS5Jx6cnF34zi/e6fYuhTi+Z/snjj5w7trU/37zYrd3adtXXzje31N59W/euTW90O1d2jMw/MP0ystrP95N7gAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQLBytxcA4IONDvR++dmj3d6io6rlvK9nvX8xZVmWUto2UPuXLzwasdSHWGk0Z5fqHTjRb/RWyiml/dsHv/zM5vv1qFXLKaXTJw6sNJrd3qU9fzV1rq3jBRbABlXOSzuGat3eYrPKUiqXOvEuTblUWi2eDuup5D2VzfrrMdLf0+0V2jbYW23reIEFsEHdnll45Ydnur1FR33x6cMDvZX/8YsLN+7Od3uXNfnSqcOjA73fe+vdueWOXsTKS6VG8/1XgP7osaPVPP/vv3x3vrNX1D7QCw/ve2B04IfnJy7fnXnfv8qzUqMVdvnquYN7D24ben3i5t9dn4ya+YHefO9qW8cLLIANqtFqzSwud3uLjmq1WimleqO5VG90e5c1abVSSml2qT67vFH+pOaW6rMb4Ndm9R3AxfpK0b/DK81mSmmp0Sj6RIv1lbaOd5M7AEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAECwcrcXAIDf8qVTh7u9Qnv+5Mlj3V7hH/zJcxtomc8+fLAzJzp1YPepA7sLPcX3bl/803aOF1gAbBTzS/XBWrXVarUKPlEpy1JKISdqtloLyytrOTLLsoGeSquVZpeX133a39JTzqt53mg2syyLnXzfP6gsZVmWmr/7dauT55brzea6/hBKpaxW+a2YWZ38e059f9r9GQgsADaK6YXlXSP9P7509a3rtwo90R+dOLq9r/b9N8fHJ6cLPdG9BnurLz57rN5ovPLLM7GTnzmw5+TesbeuT/50/Grs5H9y4vCuwf4fnLt07tZU7OR/8bFHeivlv3jr/NTCUuzkf/P8yUaz+V9/8kbs2LPnLrd1vHuwAACCCSwAgGACCwAgmMACAAgmsAAAgvkUIVCg6ujors99vttbbFa9lfKhncPd3qKjBmvVbq8AMQQWUKQsK1Uq3V5isxqsVT/7WIee07ih9JXzbq8A6yWwgAIt374z+eO/7fYWbdjzxc83l5ev/9X3u71ISinVG80b0/Pd3qKjhmrV3kp5odHs9iKwXgILYIOamlv87s/OdXuLjvrUI/uP7h4t/kHuUDg3uQMABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEExgAQAEE1gAAMEEFgBAMIEFABBMYAEABCt3ewEA2ELKeen0kf3rn5NlWX+1MrdUb6XW9r5aSmn/yFCtEvPXel7KauXy7HJ9uLcnpXR89/Z9o4Mhk1NKlbxUKeWVPE8pnTq4p95oRE1OKfWUy1lKeZadPhrwQ75XaXxbW8cLLADonFKWHdkxGjbunuwZqfWM1HrCJt9j12D/rsH+IiYfHB0qYmyWZUfG4n7IKaWUzrT5ExBYANA5jWbzF5eur3/OkbHR0b7ei7enr0/PrX/a+zz2wFhvpfz2tdvTC0vhw586uDsvZa9dulFfibx2terUoT0rzeYvxwN+wu9z4dbdto4XWADQOY1m61cTN9c/Z2ygb7Sv98rd2Tcmbq1/2vsc3bmtt1I+f+vuxJ2Z8OFPHNiZp/ytK5Pzy/Xw4acO7Wk0W7+6FPATfp9Lk9NtHe8mdwCAYAILACCYwAIACCawAACCCSwAgGACCwAgmMACAAgmsAAAggksAIBgnuQOwJaTpZRSemjPtt0jhXzF3geq5vlvTs1HnsACYMupVSoppYPbC/mm4d8vz713tCUILAC2nKWVRq1SvnD97vT8csdOWi2Xju/f3mi2OnZGukhgAbDlNFrNlNK5K3fGb7b3Db7rMVirHt+/vdUSWFuCC5UAAMFcwQJgYzk+tu3A8GChpxjq6Sl0PggsADaW0VrvaK2321vAuggsADaWdyenrk7PFnqKJx/Y1V+tFHoKtjiBBcDGcn1m/syN24We4viu7QKLQrnJHQAgmMACAAgmsAAAgrkHCwA2n3KplFJ6av+ux/aOhQ+vVcsppU8/fKDRbIYPL+d5SunLTx4p6JGrPeX8q88eDx/7l7fPt3W8wAKAzaennKeUquW8Ws4LOkVvJU+pqOF9RX7IoL8nfnhPub1kElgAsPnMLdfHUvr15Zvv3pgKH/7ZRw4O9lZ/9M7EjZn58OFfevxwOS/95esXFusr4cO//PTR+krjz3/V3tWmtXj9vettHS+wAGCzmluq35pdCB+7+o3UdxeWb83ED2+lVkrp9tzi/FI9fHhKqZlSEWvPLrb3veBucgcACOYKFgB0TqWcf+2ZEwFzcpdINjSBBQCdk/2/+9P5aBNYANA5y43Gd157Z/1zPnH4gX0jg+ufQ0EEFgB0UCvNtHm79AdaacQ/oYpA3sEFAAgmsAAAggksAIBgAgsAIJjAAgAIJrAAAIIJLACAYAILACCYwAIACCawAACCCSwAgGACCwAgmMACAAhW7vYCAHywLKWect7tLToqz7JurwAxBBbABrVjqO9rnzzR7S26oKfs3RU2PYEFsEG1Wq1mq9XtLToqy7IspUaz23vAugksgA3q1uzCf/nhr7u9RUd96qH9R3eOrjQV1odbffv4ucN7nzu8t6BTfOHkoYImp5T++fPHC5rcU86/fvpk+NjvzV7803aOF1gAsPlkWZZSWmk2G834y5zVPM+yVG80i7iGupqGyyuNIi7Prg5fWmmET15p88qqwAJgizp5aOzI3tGOna6cR95btlhfSSn97Py1Ny7fChy76ivPPDzS1/P9X49P3J4JH/7Sp05U8vzPfnp2fqkePvzrnzm5tNL4bz94M3zy2bcn2jpeYAGw5axe9dk10t/5U2+1++q2LIEFwJZza3Z+50Df+J3pC3em2n3tJw/tL2XZ/zl/qZnaS6Xh3t4n9+6cWlhs94xsRgILgC1ntYzuLCyeu3U/gZVSendyqtHmtajdg/1P7t3pAtYW4VkjAADBBBYAQDCBBQAQzD1YANA55VJpsLcaMCffWt9TuekILADohHKplFIaG+x78elj3d6FwgksAOiE1Sdg1RvNuwtL65821Futll3E2rgEFgAbxUhfT0rp4w/u/fiDRX2/3r16Ohsoq4F1a3b+z984v/5pf3js4IPbh9c/h4IILAA2it5yOaU0v1wv4vv1futElbyS5x6qTnEEFgAby/86c/Ha3blCT/H84b0n9u6ot/n1vbB2HtMAABBMYAEABBNYAADBBBYAQDCBBQAQTGABAAQTWAAAwQQWAEAwgQUAEMyT3AFgs9oxWDuyayR8bLVcSik9sK2/Vo3/usZSlqWUHhwbXl5ZCR+eUsqz7Mju+J/J1HB/W8cLLADYfPp7KimlI7tGj+waLegUJw/sLGhySunjDxX1fd7lvHT6xIHwsbMXtrW3RvgGAEDRllYaKaXbs4t355fCh+/fPljOS9fuzC0sx19kOrhzqJRlF29OF/GV3od2DTebrfGb0+GTb96db+t4gQUAm89Ko5lSOnvt9huXboUP/8qzD4/09fzyvRsTkzPhw186faJUzv/67ybml+rhw7/+uZP1RvN//mo8fPLZS5NtHe8mdwCAYAILACCYwAIACCawAACCCSwAgGACCwAgmMc0AIXIe3tTStVto3u++Plu77JmrVZKqVSt7vr857q9SkopZVnW7RWA+ySwgELkvbWUUqvZbDUb3d5lTbJSnpX+/qJ+qVLp7jLAZiewgALNjb87c/5st7dYk5ETT9R2PzD95ttz713u9i5/r9WKf8410BnuwQIACCawAACCeYsQADpnpK/39EP71z9nbKC2/iEUR2ABQCdU8lJKqVYpHxkb7fYuFE5gAbBFPbF35xN7d97fa//Vqcfu74WzS/V3b07d32vv9eD2oeFaz/rnUBCBBcAWtdxoLK20/RiRwZ5qSmlmabndF+alUl+lPLO49LfjV9t97f9vuFYVWBuZwAJgi3rr+uTPLl9r91X/+tRjpSz7s9ffbjTbe47G7sH+Lz1yuN3TsUn5FCEAQDCBBQAQTGABAAQTWAAAwQQWAEAwnyIEgM2nlGUppWN7t+8bHQwfPtBTSSl97PCuxw7sCB9ezksppdOP7m+2+THMNaqUS//4qUPhY6uT7X1vvcACgM2nr1pJKY309Yz0FfU0rLGhvoImp5T2bhsoaHIpy/btiI/ObYO9bR0vsABg85lfrqdUu3Dj7sSdmfDhTx/aXauW37h4a2puMXz48w8/kJeyvzl7dbn9p7x+qBce2bfSaP7kzJXwye9M3G7reIEFAJtPs9VKKV2fnjtzpb2/+Nfi0f1jtWr50uTMxGR8vT17dE9eys9dnZpfqocPf+GRfY1m68zl+J/J1TtzbR0vsAA2qB0Dta89d6LbW3RUtZx3ewWIIbAANqgsy3oEB2xOAgtgg7o5M/+ff/R6t7foqK8+fby/p/KJI/vqjfi7c+61+oXNWZYVeha2MoEFwEZRKZdSSqOFfSzuffoq/hKkKH63ANgollca1Tz/+XvXp+bjP7x2r4d3b9u3bXBuOf4ma1glsADYKFqtlFK6OjV77W57n9hq1+7h/n0p/lFJ8Bu+KgcAIJjAAgAIJrAAAIIJLACAYAILACCYwAIACCawAACCCSwAgGACCwAgmMACAAjmq3IAoBOylFJKOwf7v/qx4+uf1lvJ1z+E4ggsAOiEvFRKKeWlrL+n0u1dKJzAAoBOWGk2U0q3Zhf++tzl9U979tCePcMD659DQQQWAHROvdG4Nbuw/jlLK431D6E4bnIHAAjmChYAbD6VvJRS+tih3U8e3BU+vFrOU0p/ePJgs9kKH14p5ymlr3z8ofjRKaWUqpX8a585ET72Lxbea+t4gQUAm0+llKeUynmpXNinCSt5KRU2vFrYpyCzlHoKGF4ut/emn8ACgM1nvl5PKb02fuPtq7fDh3/h8T8YqlV/8Mala1Nz4cP/+PmHKnnpuz89t7C0Ej78xU8eW15pfOdH74RPfvPtq20dL7AAYLNarK/MLCyHj222Wiml+eVChqdWK6U0u1CfX6rHD0+p1UpFrL1Yby8H3eQOABDMFSwAtqi9QwPPHtjT7qtWH8j+zP49q5d51s7zRbcUgQXAFrVzoG/nQN/9vfbE7h2xy/ARI7AA2KKuz8xfm4m/ift36a+Wj+wY7djp6C6BBcAWdXV69meXrnXsdLsH+wXW1uEmdwCAYAILACCYwAIACCawAACCCSwAgGACCwAgmMACAAgmsAAAggksAIBgAgsAIJjAAgAIJrAAAIIJLACAYAILACCYwAIACFZu6+gLExOvDw4UtArwUTLQSL3Xb85fuTh/+WK3d1mTwUapZ+La3PnxhSvXu71LajWbKaXpu1Pjb7/V7V066rVsvq9aOXf28uTsQqEnGp6/uXht5OyN2+PXJws90b1m+2s/X5y6NTs/fm5i/dN+vXxncnjgzKWb49en1j/tfV6rzg32Vs+duXLj7lz48F8ML5Xz0oWzFxaXV8KH/3xHfXmlOf7Ou+GTr020+Z+y1toMDOgqAGBLGxgYWGM4rfUtwk9/+tOFbgwAsMGtPYeyVqu1xkPv3r3bbDbvdyUA2tBqtbIs6/YWwD8olUrDw8NrPLiNwAIAYC18ihAAIJjAAgAIJrAAAIIJLACAYP8X+NMc4XK1sssAAAAASUVORK5CYII=\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -w 800 -h 400\n",
    "best_hits_TM <- best_hits %>%\n",
    "\tfilter(component == \"microcins.verified\") %>%\n",
    "\tfilter(hmmerHit == TRUE) %>%\n",
    "\tgroup_by(pephash,verified,sample) %>% summarise(n = n()) %>%\n",
    "\tungroup() %>%\n",
    "\tmutate(verified = as.factor(verified)) %>%\n",
    "#Uncomment row below to include all 1224 assemblies including those without HMM-verified microcin hits\n",
    "#\tadd_row(pephash=NA, verified=\"no_microcin\",n=samples_wo_microcins) %>% \n",
    "\ttreemap(\n",
    "\t\tindex = c(\"verified\",\"pephash\"),\n",
    "        vSize = \"n\",\n",
    "\t\tborder.col=c(\"black\",\"white\"),\n",
    "    \tborder.lwds=c(3,2.5),\n",
    "\t\tfontsize.labels=c(0,0),\n",
    "\t\tpalette = c(\"#b6d9e8\",\"#ffd1d1\")\n",
    "\t)\n",
    "\n",
    "# Uncomment below to save\n",
    "#ggsave(\"plots/AF/03_Sankey_plot_FINAL.svg\", width = 4, height = 2, units = c(\"in\"))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Data for table\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "best_hits %>%\n",
    "\tfilter(component == \"microcins.verified\") %>%\n",
    "\tfilter(verified == FALSE) %>%\n",
    "\tfilter(hmmerHit == TRUE) %>%\n",
    "\tcount(seq, sseqid, pident) %>%\n",
    "\twrite_csv(\"data/supplemental/Ecoli_microcin_novel.csv\")"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "184da163bdcbc68ba00985bd6e7412525ff7507797be2878afd169be169c48f1"
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
